File: android/src/main/java/expo/modules/logisticsscanner/scanner/ScannerProvider.kt
package expo.modules.logisticsscanner.scanner

import android.graphics.SurfaceTexture

interface ScannerProvider {
  fun start(
    onScan: (ScanResult) -> Unit,
    onStarted: () -> Unit = {},
    onFailed: (ScanError) -> Unit = {},
    onError: (ScanError) -> Unit = {},
  )
  fun stop()
  fun setTorch(enabled: Boolean)
  fun setPreviewSurfaceTexture(texture: SurfaceTexture?)
}

File: ios/scanner/ScannerProvider.swift
import Foundation

protocol ScannerProvider: AnyObject {
  func start(
    onScan: @escaping (ScanResult) -> Void,
    onStarted: @escaping () -> Void,
    onFailed: @escaping (ScanError) -> Void,
    onError: @escaping (ScanError) -> Void
  )
  func stop()
  func setTorch(enabled: Bool)
}
